### ISSUE TITLE
Doubutsu no Mori - FPS drops when saving and loading

### ISSUE THREAD

**Issue by GhytricsStudio** [simple64/simple64#380]
Hello,

In the japanese game Doubutsu no Mori, every time a save file is read or written, there is a huge FPS drop.
This has been a problem since at least Mupen64 0.5

**More informations**
- I'm using the original ROM (without any English patches)
- The game's save file is a 128 Ko .fla file.
- The game make use of the Controller Pak for storing letters at the post office (where reading/writing the mpk file also cause a FPS drop)
- This is the only game where I experience such issue

**My specs**
- Windows 10 latest version
- CPU: AMD Ryzen 7 1700 @ 3.5GHz
- GPU: NVIDIA GeForce GTX 1080
- RAM: 16 Go

**Comment by loganmc10** [comment]
Just to be sure, are you playing the original Japenese version, or the English translation?

**Comment by GhytricsStudio** [comment]
I'm playing the original Japanese version

**Comment by loganmc10** [comment]
ok, and have you seen how it works on a real console? Are you sure the slowdown doesn't happen there? It's a bit hard for me to test since I don't understand Japanese (I don't know how to save)

**Comment by GhytricsStudio** [comment]
Yes, there is no slowdown on a real console. I found a video to demonstrate:
https://youtu.be/D8Z4NkVEgE4
At 0:34, you can see the save loading with a nice jingle. In the emulator, it's stuttering.
There's no problem on Project64 for example.

I understand the difficulties of testing a Japanese game. I suppose it could be reproduced with a translated ROM.

At the end of the video, you can see of the player initiated the saving of the game.

**Comment by loganmc10** [comment]
Can you share your save file(s) so I can test the loading? That seems like the quickest way to test

**Comment by GhytricsStudio** [comment]
No problem, I will provide one tomorrow!

**Comment by GhytricsStudio** [comment]
Hello,
Here is a fresh new save file.
To load, simply always press A all the time. To save, talk to the little character next to the house and select the third option.
Thank you!

[Doubutsu no Mori save.zip](https://github.com/simple64/simple64/files/12188178/Doubutsu.no.Mori.save.zip)

**Comment by loganmc10** [comment]
I don't think I can reproduce the issue, see my video here: https://www.youtube.com/watch?v=POWg-W1jQQ4

Sorry for the weird window size, but it doesn't seem to freeze when I save or load the game

**Comment by GhytricsStudio** [comment]
Very interesting, it indeed works very well.
I always had this problem since 2007 only on Mupen-based emulators and I never could find why this happened...

**Comment by loganmc10** [comment]
You are using simple64, correct?

Perhaps it is an anti-virus that kicks in when the save file is read/written?

**Comment by GhytricsStudio** [comment]
Yes, I'm using Simple64, the latest version, with no tweaks of the configs.
I have the default Windows 10 antivirus, I'll try to disable it
What bothers me is that it's the only game with this issue
Could this be because of an AMD CPU ?

**Comment by GhytricsStudio** [comment]
Oh, that may be a Windows issue. I don't have a Linux to test it, unfortunately.
Here is a video I made about how it behaves to me: https://youtu.be/x44ZLsG3KOM

**Comment by DarthSidious666** [comment]
It sounds to me as though your CPU is struggling to keep the pacing.
Although a Ryzen 7 1700 can run the emulator it's getting quite old and the more accurate the emulation becomes the more strain it will place on the CPU.

It's not on my list of issues for J games otherwise I would have made a note of this.

**Comment by GhytricsStudio** [comment]
I don't think it's because of the CPU. I played this game on 3 differents PC between 2008 and now: it really happens to me only in this very game and only during saving/loading.
I will try this at a friend's house who has a Windows PC and a recent Intel CPU. I should also try it on a Linux VM to see if it could be OS-related.

Could you try this game with the save file I provided, please?

**Comment by DarthSidious666** [comment]
I do have a slight slowdown when saving and loading,
It's really minor though like 1-2 seconds. But I'll add it to the list.
